src/error_reporter.rs

```diff
@@ -241,6 +241,91 @@
 //!     println!("{}", report);
 //! }
 //! ```
+//!
+//! It also is not required that the wrapped error be `'static`. However, if the inner reference
+//! doesn't itself implement `Error`, then it can't be printed by `std::Report` via
+//! `Error::source`. In this situation, use `Display` instead:
+//!
+//! ```rust
+//! # use std::fmt;
+//! # use std::error::Error;
+//!
+//! # use trial_and_error::Report;
+//!
+//! #[derive(Debug)]
+//! struct SuperError<'a> {
+//!     side: &'a str
+//! }
+//!
+//! impl<'a> fmt::Display for SuperError<'a> {
+//!     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+//!         write!(f, "SuperError is here: {}", self.side)
+//!     }
+//! }
+//!
+//! impl<'a> Error for SuperError<'a> {}
+//!
+//! fn main() {
+//!     let msg = String::from("The source of the error");
+//!     let mut report = Report::new(SuperError { side: &msg });
```

Comment:
I feel like calling this a source is just going to confuse people.

```suggestion
//!     let msg = String::from("Huzzah!");
//!     let mut report = Report::new(SuperError { side: &msg });
```